webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/DataSourceController.scala

```diff
@@ -81,51 +83,52 @@ class DataSourceController @Inject()(
-                val id = DataSourceId(name, organization)
-                val resumableUploadInformation = ResumableUploadInformation(chunkSize, totalChunkCount)
-                for {
-                  _ <- if (!uploadService.isKnownUpload(uploadId))
-                    webKnossosServer.validateDataSourceUpload(id) ?~> "dataSet.upload.validation.failed"
-                  else Fox.successful(())
-                  chunkFile <- request.body.file("file") ?~> "zip.file.notFound"
-                  _ <- uploadService.handleUploadChunk(uploadId,
-                                                       id,
-                                                       resumableUploadInformation,
-                                                       chunkNumber,
-                                                       new File(chunkFile.ref.path.toString))
-                } yield {
-                  Ok
-                }
-            }
-          )
+      accessTokenService.validateAccess(UserAccessRequest.administrateDataSources) {
+        AllowRemoteOrigin {
+          uploadForm
+            .bindFromRequest(request.body.dataParts)
+            .fold(
+              hasErrors = formWithErrors => Fox.successful(JsonBadRequest(formWithErrors.errors.head.message)),
+              success = {
+                case (name, organization, chunkNumber, chunkSize, totalChunkCount, uploadId) =>
+                  val id = DataSourceId(name, organization)
+                  val resumableUploadInformation = ResumableUploadInformation(chunkSize, totalChunkCount)
+                  for {
+                    _ <- if (!uploadService.isKnownUpload(uploadId))
+                      webKnossosServer.validateDataSourceUpload(id) ?~> "dataSet.upload.validation.failed"
+                    else Fox.successful(())
+                    chunkFile <- request.body.file("file") ?~> "zip.file.notFound"
+                    _ <- uploadService.handleUploadChunk(uploadId,
+                                                         id,
+                                                         resumableUploadInformation,
+                                                         chunkNumber,
+                                                         new File(chunkFile.ref.path.toString))
+                  } yield {
+                    Ok
+                  }
+              }
+            )
+        }
       }
-    }
   }
 
-  def finishUpload = Action.async(validateJson[UploadInformation]) { implicit request =>
+  def finishUpload: Action[UploadInformation] = Action.async(validateJson[UploadInformation]) { implicit request =>
     accessTokenService.validateAccess(UserAccessRequest.administrateDataSources) {
       AllowRemoteOrigin {
         for {
-          (dataSourceId, initialTeams) <- uploadService.finishUpload(request.body)
+          (dataSourceId, initialTeams, needsConversion) <- uploadService.finishUpload(request.body)
```

Comment:
Do we use the `needsConversion` flag here? I think it can be silenced with `_`.

Reply:
Good catch! That was a leftover from my attempts at #5001 which apparently slipped into this branch. I removed the additional return value.